examples/light-switch-app/nrfconnect/main/LightSwitch.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "LightSwitch.h"
+#include "AppEvent.h"
+#include "BindingHandler.h"
+
+#include <app/server/Server.h>
+#include <app/util/binding-table.h>
+#include <controller/InvokeInteraction.h>
+
+using namespace chip;
+using namespace chip::app;
+
+LightSwitch LightSwitch::sLightSwitch;
+
+void LightSwitch::Init(chip::EndpointId aLightSwitchEndpoint)
+{
+    BindingHandler::Init();
+    mLightSwitchEndpoint = aLightSwitchEndpoint;
+}
+
+void LightSwitch::InitiateActionSwitch(Action mAction)
+{
+    BindingHandler::BindingData * data = Platform::New<BindingHandler::BindingData>();
+    if (data)
+    {
+        data->EndpointId = mLightSwitchEndpoint;
+        data->ClusterId  = Clusters::OnOff::Id;
+        switch (mAction)
+        {
+        case Action::Toggle:
+            data->CommandId = Clusters::OnOff::Commands::Toggle::Id;
+            break;
+        case Action::On:
+            data->CommandId = Clusters::OnOff::Commands::On::Id;
+            break;
+        case Action::Off:
+            data->CommandId = Clusters::OnOff::Commands::Off::Id;
+            break;
+        default:
+            Platform::Delete(data);
+            return;
+        }
+        DeviceLayer::PlatformMgr().ScheduleWork(BindingHandler::SwitchWorkerHandler, reinterpret_cast<intptr_t>(data));
```

Comment:
And who is going to delete the `data`? Can we assume that the `PlatformMgr` does some cleanup?

Reply:
I checked that - PlatformMgr provides only memory allocation in a "New" method so You are right - we should delete it.

Reply:
If we were to delete it it would be rather tricky as here we only schedule some asynchronous work, so we would know know when to delete the resources. So good that the ownership is taken, pity it happens in quite obscure manner :)